src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -73,7 +74,22 @@ private Database Database
                         {
                             // Reuse existing connection 
                             Server server = new Server(this.serverConnection);
-                            this.database = new Database(server, this.serverConnection.DatabaseName);
+                            // The default database name is the database name of the server connection
+                            string dbName = this.serverConnection.DatabaseName;                          
+                            if (this.connectionInfo != null)
+                            {
+                                // If there is a query DbConnection, use that connection to get the database name
+                                // This is preferred since it has the most current database name (in case of database switching)
```

Comment:
Looking at this again, this seems to be more of a flaw in the ServerConnection passed into PeekDefinition's constructor than it is for PeekDefinition. It seems like the Database name for the BindingQueue's ServerConnection will never be updated. Updating the database name for the binding queue [in the connection service's database changed callback](https://github.com/Microsoft/sqltoolsservice/blob/dev/src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs#L942) might be a better solution than checking for the database name of a Query connection.